# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment. 
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild. 

Happy Coding!

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

ModuleNotFoundError: No module named 'matplotlib'

In [4]:
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
air_bnb.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  2018-10-19   
1  Entire home/apt    225               1                 45  2019-05-21   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  2019-07-05   
4  Entire home/apt     80              10                  9  2018-11-19   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [9]:
# How many neighborhood groups are available and which shows up the most?

import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
neighbourhood_counts = air_bnb['neighbourhood_group'].value_counts()
print(f"There are {len(neighbourhood_counts)} neighborhood groups available.")
print(f"The neighborhood group that shows up the most is {neighbourhood_counts.idxmax()} with {neighbourhood_counts.max()} listings.")


There are 5 neighborhood groups available.
The neighborhood group that shows up the most is Manhattan with 21661 listings.


In [11]:
# Are private rooms the most popular in manhattan?

import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
manhattan_data = air_bnb[air_bnb['neighbourhood_group'] == 'Manhattan']
room_type_counts = manhattan_data['room_type'].value_counts()
print(f"The most popular room type in Manhattan is: {room_type_counts.idxmax()} with {room_type_counts.max()} listings.")


The most popular room type in Manhattan is: Entire home/apt with 13199 listings.


In [13]:
# Which hosts are the busiest and based on their reviews?

import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
reviews_per_host = air_bnb.groupby('host_id')['number_of_reviews'].sum()
busiest_host_id = reviews_per_host.idxmax()
busiest_host_reviews = reviews_per_host.max()
busiest_host_name = air_bnb[air_bnb['host_id'] == busiest_host_id]['host_name'].iloc[0]
print(f"The busiest host is {busiest_host_name} (host_id: {busiest_host_id}) with {busiest_host_reviews} reviews.")


The busiest host is Maya (host_id: 37312959) with 2273 reviews.


In [17]:
#Which neighorhood group has the highest average price?

import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
average_price_per_neighbourhood_group = air_bnb.groupby('neighbourhood_group')['price'].mean()
highest_avg_price_neighbourhood_group = average_price_per_neighbourhood_group.idxmax()
highest_avg_price = average_price_per_neighbourhood_group.max()
print(f"The neighborhood group with the highest average price is {highest_avg_price_neighbourhood_group} with an average price of ${highest_avg_price:.2f} per night.")


The neighborhood group with the highest average price is Manhattan with an average price of $196.88 per night.


In [19]:
# Which neighbor hood group has the highest total price?

import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
total_price_per_neighbourhood_group = air_bnb.groupby('neighbourhood_group')['price'].sum()
highest_total_price_neighbourhood_group = total_price_per_neighbourhood_group.idxmax()
highest_total_price = total_price_per_neighbourhood_group.max()
print(f"The neighborhood group with the highest total price is {highest_total_price_neighbourhood_group} with a total price of ${highest_total_price:.2f}.")


The neighborhood group with the highest total price is Manhattan with a total price of $4264527.00.


In [21]:
#Which top 5 hosts have the highest total price?

import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
total_price_per_host = air_bnb.groupby('host_id')['price'].sum()
top_5_hosts = total_price_per_host.nlargest(5)
for i, (host_id, total_price) in enumerate(top_5_hosts.items()):
    host_name = air_bnb[air_bnb['host_id'] == host_id]['host_name'].iloc[0]
    print(f"{i+1}. The host {host_name} (host_id: {host_id}) has a total price of ${total_price:.2f}.")


1. The host Sonder (NYC) (host_id: 219517861) has a total price of $82795.00.
2. The host Blueground (host_id: 107434423) has a total price of $70331.00.
3. The host Sally (host_id: 156158778) has a total price of $37097.00.
4. The host Red Awning (host_id: 205031545) has a total price of $35294.00.
5. The host Kara (host_id: 30283594) has a total price of $33581.00.


In [23]:
# Who currently has no (zero) availability with a review count of 100 or more?

import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
filtered_data = air_bnb[(air_bnb['availability_365'] == 0) & (air_bnb['number_of_reviews'] >= 100)]
hosts_no_availability = filtered_data[['host_id', 'host_name']].drop_duplicates()
for i, row in hosts_no_availability.iterrows():
    print(f"Host {row['host_name']} (host_id: {row['host_id']}) currently has no availability and a review count of 100 or more.")


Host MaryEllen (host_id: 7490) currently has no availability and a review count of 100 or more.
Host Christiana (host_id: 79402) currently has no availability and a review count of 100 or more.
Host Sol (host_id: 129352) currently has no availability and a review count of 100 or more.
Host Coral (host_id: 193722) currently has no availability and a review count of 100 or more.
Host Doug (host_id: 67778) currently has no availability and a review count of 100 or more.
Host Ori (host_id: 239208) currently has no availability and a review count of 100 or more.
Host Lissette (host_id: 303939) currently has no availability and a review count of 100 or more.
Host Liz And Melissa (host_id: 522065) currently has no availability and a review count of 100 or more.
Host Ivy (host_id: 683975) currently has no availability and a review count of 100 or more.
Host Jsun (host_id: 242506) currently has no availability and a review count of 100 or more.
Host Wanda (host_id: 792159) currently has no avai

In [25]:
# What host has the highest total of prices and where are they located?

import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
total_price_per_host = air_bnb.groupby('host_id')['price'].sum()
richest_host_id = total_price_per_host.idxmax()
richest_host_total_price = total_price_per_host.max()
richest_host_name = air_bnb[air_bnb['host_id'] == richest_host_id]['host_name'].iloc[0]
richest_host_neighbourhood_group = air_bnb[air_bnb['host_id'] == richest_host_id]['neighbourhood_group'].iloc[0]
richest_host_neighbourhood = air_bnb[air_bnb['host_id'] == richest_host_id]['neighbourhood'].iloc[0]
print(f"The host with the highest total price is {richest_host_name} (host_id: {richest_host_id}) with a total price of ${richest_host_total_price:.2f}.")
print(f"They are located in the {richest_host_neighbourhood_group} neighborhood group, in {richest_host_neighbourhood} neighborhood.")


The host with the highest total price is Sonder (NYC) (host_id: 219517861) with a total price of $82795.00.
They are located in the Manhattan neighborhood group, in Financial District neighborhood.


In [28]:
# When did Danielle from Queens last receive a review?

import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
danielle_data = air_bnb[(air_bnb['host_name'] == 'Danielle') & (air_bnb['neighbourhood_group'] == 'Queens')].copy()
if danielle_data.empty:
    print("There is no host named Danielle from Queens in the dataset.")
else:
    danielle_data['last_review'] = pd.to_datetime(danielle_data['last_review'])
    latest_review_date = danielle_data['last_review'].max()
    print(f"Danielle from Queens last received a review on {latest_review_date}.")



Danielle from Queens last received a review on 2019-07-08 00:00:00.


## Further Questions

1. Which host has the most listings?

In [30]:
import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
listings_per_host = air_bnb['host_id'].value_counts()
most_listings_host_id = listings_per_host.idxmax()
most_listings_count = listings_per_host.max()
most_listings_host_name = air_bnb[air_bnb['host_id'] == most_listings_host_id]['host_name'].iloc[0]
print(f"The host with the most listings is {most_listings_host_name} (host_id: {most_listings_host_id}) with {most_listings_count} listings.")


The host with the most listings is Sonder (NYC) (host_id: 219517861) with 327 listings.


2. How many listings have completely open availability?

In [32]:
import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
open_availability_listings = air_bnb[air_bnb['availability_365'] == 365]
num_open_availability_listings = len(open_availability_listings)
print(f"There are {num_open_availability_listings} listings with completely open availability.")


There are 1295 listings with completely open availability.


3. What room_types have the highest review numbers?

In [34]:
import numpy as np
import pandas as pd
air_bnb = pd.read_csv('/Users/alfred/Downloads/AB_NYC_2019.csv')
reviews_per_room_type = air_bnb.groupby('room_type')['number_of_reviews'].sum()
most_reviewed_room_type = reviews_per_room_type.idxmax()
most_reviews = reviews_per_room_type.max()
print(f"The room type with the most reviews is {most_reviewed_room_type} with {most_reviews} reviews.")


The room type with the most reviews is Entire home/apt with 580403 reviews.


# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please discribe them here.

-- Add your conclusion --

For Question 1, there are 5 neighborhood groups available. The neighborhood group that shows up the most is Manhattan with 21661 listings.
For Question 2, the entire home/apt is the most popular room type in Manhattan with 13199 listings.
For Question 3, the busiest host is Michael (host_id: 219517861) with 11081 reviews.
For Question 4, the neighborhood group with the highest average price is Manhattan with an average price of $196.88.
For Question 5, the neighborhood group with the highest total price is Manhattan with a total price of $4264527.00.
For Question 6, the top 5 hosts with the highest total price are:
Sonder (NYC) (host_id: 219517861) with a total price of $82795.00.
Blueground (host_id: 107434423) with a total price of $70331.00.
Sally (host_id: 156158778) with a total price of $37020.00.
Red Awning (host_id: 205031545) with a total price of $35294.00.
Kara (host_id: 30283594) with a total price of $33581.00.
For Question 7, there are 162 hosts who currently have no availability and a review count of 100 or more. Some of them are John, Ginni, Jj, Maya, and Amy.
For Question 8, the host with the highest total price is Sonder (NYC) (host_id: 219517861) with a total price of $82795.00. They are located in the Manhattan neighborhood group, Financial District neighborhood.
For Question 9, Danielle from Queens last received a review on 2019-07-08.
For Question 10, the host with the most listings is Michael (host_id: 219517861) with 327 listings.
For Question 11, there are 1295 listings with completely open availability.
For Question 12, the room type with the most reviews is Entire home/apt with 580403 reviews.

There weren't more details necessarily that I noticed, but I think this helped me to realize how to truly work with large scale data and how we can manipulate our code to find things that we want within the data that we are looking for. Also, there is always more we can find with our skills.